src/WordsFilter.java

```diff
@@ -0,0 +1,78 @@
+import java.util.Arrays;
+import java.util.Scanner;
+
+public class WordsFilter {
+    private static Scanner scanner = new Scanner(System.in);
+
+    public static void main(String[] args) {
+        System.out.println("Input words using comma: ");
+        String userInput = scanner.nextLine();
+        String[] splitInput = userInput.split(",");
+        int input = 5;
+        while (input != 0) {
+            System.out.println("Choose a filter:");
+            System.out.println("1 - Max length");
+            System.out.println("2 - Starts with");
+            System.out.println("3 - Ends with");
+            System.out.println("4 - Contains");
+            System.out.println("0 - Exit");
+            input = scanner.nextInt();
+            scanner.nextLine();
+            if (input == 1) {
+                System.out.println("Please, provide a maximum length");
+                int providedLength = scanner.nextInt();
+                scanner.nextLine();
+                String[] newArray = new String[splitInput.length + 1];
```

Comment:
try not to create a new array but use your old one and loop through it to determine if it has required length, cause when you run it it returns one extra item in list with null value

Reply:
the same for all filters